app/models/learning_unit_succession.rb

```diff
@@ -3,14 +3,23 @@ class LearningUnitSuccession < ApplicationRecord
   belongs_to :successor, class_name: 'LearningUnit'
   belongs_to :cycle
 
-  validate :different_learning_units_for_succession
   validates :cycle, presence: true
   validates :predecessor, presence: true
   validates :successor, presence: true
+  validate :different_learning_units_for_succession,
+           :learning_units_must_belong_to_cycle
 
   def different_learning_units_for_succession
     return unless predecessor == successor
 
     errors.add(:successor, 'Successor has to be different than predecessor')
   end
+
+  def learning_units_must_belong_to_cycle
+    return if (
```

Comment:
un poco raro esto imho. 

Primero, `cycle.present? ? cycle.learning_units : []` se puede convertir en `cycle&.learning_units.presence || []`

entonces quizás algo así:

```
cycle_learning_units = cycle&.learning_units.presence || []
xxx = [predecessor, successor] - cycle_learning_units
return if xxx.empty?

```

Ahí quizás qué el nombre de XXX permite de explicitar el caso borde que protege este early return

Reply:
There I fixed it to use the .presence operator (didn't know about it)